tests/ui/double_must_use.rs

```diff
@@ -21,6 +21,17 @@ pub fn must_use_with_note() -> Result<(), ()> {
     unimplemented!();
 }
 
+// vvvv Should not lint (#10486)
+#[must_use]
+async fn async_must_use() -> usize {
+    unimplemented!();
+}
+
+#[must_use]
+async fn async_must_use_result() -> Result<(), ()> {
```

Comment:
It looks like that this case is false negative. It would be better to check that the return type in async function already has `#[must_use]`. I think we can get the type from `Future::Output` like below:
```rs
let infcx = cx.tcx.infer_ctxt().build();
let Some(ret_ty) = infcx.get_impl_future_output_ty(return_ty(cx, item_id));
```